src/System.Diagnostics.Tracing.Telemetry/src/System/Diagnostics/Tracing/TelemetryListener.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// A TelemetryListener something that forwards on events written with TelemetrySource.
+    /// It is an IObservable (has Subscribe method), and it also has a Subscribe overload that
+    /// lets you specify a 'IsEnabled' precicate that users of TelemetrySource will use for 
+    /// 'quick checks'.   
+    /// 
+    /// The item in the stream is a KeyValuePair<string, object> where the string is the name
+    /// of the telemetry item and the object is the payload (typically an anonumous type).  
+    /// 
+    /// There may be many TelemetryListeners in the system, but we encourage the use of
+    /// The TelemetrySource.DefaultSource which goes to the TelemetryListener.DefaultListener.
+    /// 
+    /// If you need to see 'everything' you can subscribe to the 'AllListeners' event that
+    /// will fire for every live TelemetryListener in the appdomain (past or present). 
+    /// </summary>
+    public class TelemetryListener : TelemetrySource, IObservable<KeyValuePair<string, object>>, IDisposable
+    {
+        /// <summary>
+        /// When a TelemetryListener is created it is given a name.   Return this.  
+        /// </summary>
+        public string Name { get; private set; }
+        /// <summary>
+        /// This is the TelemetryListener that is used by default by the class library.   
+        /// Generally you don't want to make your own but rather have everyone use this one, which
+        /// insures that everyone who wished to subscribe gets the callbacks.  
+        /// The main reason not to us this one is that you WANT isolation from other 
+        /// events in the system (e.g. multi-tenancy).  
+        /// </summary>
+        public static TelemetryListener DefaultListener { get { return s_default; } }
+
+        /// <summary>
+        /// When you subscribe to this you get callbacks for all NotificationListeners in the appdomain
+        /// as well as those that occured in the past, and all future Listeners created in the future. 
+        /// </summary>
+        public static event Action<TelemetryListener> AllListeners
```

Comment:
I am not a big fan of EventHandler.   In theory it is there for extensibility (you can add more 'arguments' to the event), but I can't point to a single instance where we actually took advantage of that extensibility.    Also the 'sender' parameter of EventHandlers seems to cause more confusion than value.   

Thus I think it is reasonable to drop the pattern if there is a low probability that it will add value, and there is no obvious standard to align to.   In this case I can't imagine another piece of information for this event (its purpose is simply to indicate what TelemetryListeners exist), so I think it is OK to dispense with it.

That being said, I think it is  a minor point and I am willing to change it if someone can make a clear case for value.